lib/penelope/nlp/pos_tagger.ex

```diff
@@ -0,0 +1,96 @@
+defmodule Penelope.NLP.POSTagger do
+  @moduledoc """
+  The part-of-speech tagger transforms a tokenized sentence into a list of
+  `{token, pos_tag}` tuples. The tagger takes no responsibility for
+  tokenization; this means that callers must be careful to maintain the same
+  tokenization scheme between training and evaluating to ensure the best
+  results.
+
+  As this tagger does not ship with a pretrained model, it is both
+  language- and tagset-agnostic, though the default feature set used
+  (see `POSFeaturizer`) was designed for English.
+
+  See `POSTaggerTrainer.train/2` for an example
+  of how to train a new POS tagger model.
+  """
+
+  alias Penelope.ML.Pipeline
+
+  @type model :: %{
+          tagger: [{atom, any}],
+          featurizer: [{atom, any}]
+        }
+
+  @doc """
+  Fits the tagger model. The following keys may be fed to `pipelines`
+  to include custom components in the tagger pipeline:
+
+  |key               |default                       |
+  |------------------|------------------------------|
+  |`featurizer`      |`[{:pos_featurizer, []}]`     |
+  |`tagger`          |`[{:crf_tagger, []}]`         |
+  """
+  @spec fit(
+          context :: map,
+          x :: [tokens :: [String.t()]],
+          y :: [tags :: [String.t()]],
+          pipelines :: keyword
+        ) :: model
+  def fit(context, x, y, pipelines \\ []) do
```

Comment:
Why not build a single pipeline containing the featurizer and tagger?

Reply:
Because I was following the example of `IntentClassifier` too closely.

Reply:
Also to allow users to create their own featurizer pipeline.